### ISSUE TITLE
Support structural interface implementation via Extension Class without having to manually create an IProxyFactory service impl

### ISSUE THREAD

**Issue by rsmckinney** [manifold-systems/manifold#337]
Currently, if an extension class implements a structural interface with extension methods, an IProxyFactory service implementation must be provided as a proxy for the extended class to the structural interface. Although documented, this is not at all obvious. It would be much better if the IProxyFactory implementation were automatically generated and registered at compile-time.

**Comment by rsmckinney** [comment]
*Problem description:*

```java
@Structural
public interface Numeric {
  double doubleValue();
}

@Extension
public abstract class MyStringExt implements Numeric {
  public static double doubleValue(@This String thiz) {
    return Double.parseDouble(thiz);
  }
}

public class Test {
  public void testStructuralCallViaExtension() {
    Numeric num = "3.14";
    double d = num.doubleValue();
    assertEquals(3.14, d);
  }
}
```
Currently, for this example to work we must create an `IProxyFactory` implementation and register it in a `META-INF/services/..IProxyFactory` file for unnamed module projects and for named module projects register a provider in the `module-info.java` source file.  Although this is documented, it is not something one would expect, thus someone attempting to use structural interfaces implemented with extension classes may abandon the effort before it is understood, which is too bad because it offers a powerful solution to problems that otherwise may suffer from lesser remedies.

*Proposed solution:*

To resolve the issue we will generate an `IProxyFactory_gen` implementation at compile-time for every interface an extension class implements. We must also generate a `META-INF/services/..IProxyFactory_gen` file for unnamed module projects. Note, `IProxyFactory_gen` will be a new interface that extends `IProxyFactory` to avoid having to reconcile changes with manually implemented `IProxyFactory` providers. Since all artifacts are generated exclusively to the project's *target* directory, the `IProxyFactory_gen` solution will be largely invisible during development.

For named-module projects we'll use the same, quietly generated `META-INF/services/..IProxyFactory_gen` file by implementing our own Provider `Iterator` and an ever so tiny bit of internal API abuse on the `ServiceLoader`.  Otherwise, the named-module solution would result in an unjustifiable tower of JPMS madness that I'm not quite sure would work.

**Comment by rsmckinney** [comment]
Feature implemented with release 2022.1.1
